src/qat/purr/backends/utilities.py

```diff
@@ -204,12 +205,24 @@ def eval(self, x: np.ndarray) -> np.ndarray:
         return coef * (gauss - zae_chunk)
 
 
-class Sech(ComplexFunction):
+class SechFunction(ComplexFunction):
+    """
+    Implements a sech pulse defined by sech(x / width). Note that it is not normalized to be
+    zero at the edges.
+    """
+
     def __init__(self, width):
         self.width = width
 
     def eval(self, x: np.ndarray) -> np.ndarray:
-        return 1 / cosh(x / self.width)
+        # Having a narrow width can cause overflows in numpy
+        # Restricting the argument such that cosh is within the max float range
+        # will overcome this, and has a neglibable effect (as sech(x) outside this
+        # range is practically zero).
+        x = np.array(
+            [max(min(np.real(val) / self.width, MAX_COSH_ARG), -MAX_COSH_ARG) for val in x]
+        )
+        return 1 / np.cosh(x)
```

Comment:
How about this

`np.clip(x.real / width, -MAX_COSH_ARG, MAX_COSH_ARG)`

Timed in a notebook, It's much faster